sklearn/feature_selection/tests/test_sequential.py

```diff
@@ -41,21 +42,98 @@ def test_n_features_to_select(direction, n_features_to_select):
         cv=2,
     )
     sfs.fit(X, y)
-    if n_features_to_select is None:
-        n_features_to_select = 5  # n_features // 2
+
     assert sfs.get_support(indices=True).shape[0] == n_features_to_select
     assert sfs.n_features_to_select_ == n_features_to_select
     assert sfs.transform(X).shape[1] == n_features_to_select
 
 
+@pytest.mark.parametrize("direction", ("forward", "backward"))
+@pytest.mark.parametrize("tol", (1e-3, None))
+def test_n_features_to_select_auto(direction, tol):
```

Comment:
> Since we are raising a warning for the deprecation, we should force n_features_to_select="auto" and tol=None in the tests. 

There are tests that explicitly takes `"auto"` as an argument, is that insufficient?

Reply:
yes it would be enough in this case

Reply:
force n_features_to_select="auto" with https://github.com/scikit-learn/scikit-learn/pull/20145/commits/8f2b71918369f342b5a758800a09e1f5d709dedf